receiver/jaegerreceiver/factory.go

```diff
@@ -85,14 +86,33 @@ func (f *Factory) CustomUnmarshaler() component.CustomUnmarshaler {
 		if len(protocols) == 0 {
 			return fmt.Errorf("must specify at least one protocol when using the Jaeger receiver")
 		}
-		for k := range protocols {
-			if _, ok := receiverCfg.Protocols[k]; !ok {
-				if receiverCfg.Protocols[k], err = defaultsForProtocol(k); err != nil {
-					return err
-				}
-			}
+		// UnmarshalExact will ignore empty entries like a protocol with no values, so if a typo happened
+		// in the protocol that is intended to be enabled will not be enabled. So check if the protocols
+		// include only known protocols.
```

Comment:
I would move the current comment down to line 113 and put a different comment here that says something like "UnmarshalExact will not set struct properties to nil even if no key is provided, so set the protocol structs to nil where the keys were ommitted"

Also the comment on line 83 is wrong now

Reply:
Done.